integration-test/src/test/java/org/apache/iotdb/db/it/aggregation/IoTDBCountIfIT.java

```diff
@@ -0,0 +1,181 @@
+
+    // keep < threshold
+    expectedHeader =
+        new String[] {
+          "Count_if(root.db.d1.s1 = 0 & root.db.d1.s2 = 0, keep < 3, \"ignoreNull\"=\"true\")",
+          "Count_if(root.db.d1.s1 = 1 & root.db.d1.s2 = 0, keep < 3)"
+        };
+    retArray = new String[] {"0,0,"};
+    resultSetEqualTest(
+        "select Count_if(s1=0 & s2=0, keep<3, 'ignoreNull'='true'), Count_if(s1=1 & s2=0, keep<3) from root.db.d1",
+        expectedHeader,
+        retArray);
+  }
+
+  @Test
+  public void testCountIfRespectNull() {
+    // threshold constant
+    String[] expectedHeader =
+        new String[] {
+          "Count_if(root.db.d1.s1 = 0 & root.db.d1.s2 = 0, 3, \"ignoreNull\"=\"false\")",
+          "Count_if(root.db.d1.s1 = 1 & root.db.d1.s2 = 0, 3, \"ignoreNull\"=\"false\")"
+        };
+    String[] retArray = new String[] {"1,1,"};
+    resultSetEqualTest(
+        "select Count_if(s1=0 & s2=0, 3, 'ignoreNull'='false'), Count_if(s1=1 & s2=0, 3, 'ignoreNull'='false') from root.db.d1",
+        expectedHeader,
+        retArray);
+
+    // keep >= threshold
+    expectedHeader =
+        new String[] {
+          "Count_if(root.db.d1.s1 = 0 & root.db.d1.s2 = 0, keep >= 3, \"ignoreNull\"=\"false\")",
+          "Count_if(root.db.d1.s1 = 1 & root.db.d1.s2 = 0, keep >= 3, \"ignoreNull\"=\"false\")"
+        };
+    resultSetEqualTest(
+        "select Count_if(s1=0 & s2=0, keep>=3, 'ignoreNull'='false'), Count_if(s1=1 & s2=0, keep>=3, 'ignoreNull'='false') from root.db.d1",
+        expectedHeader,
+        retArray);
+
+    // keep < threshold
+    expectedHeader =
+        new String[] {
+          "Count_if(root.db.d1.s1 = 0 & root.db.d1.s2 = 0, keep < 3, \"ignoreNull\"=\"false\")",
+          "Count_if(root.db.d1.s1 = 1 & root.db.d1.s2 = 0, keep < 3, \"ignoreNull\"=\"false\")"
+        };
+    retArray = new String[] {"1,0,"};
+    resultSetEqualTest(
+        "select Count_if(s1=0 & s2=0, keep<3, 'ignoreNull'='false'), Count_if(s1=1 & s2=0, keep<3, 'ignoreNull'='false') from root.db.d1",
+        expectedHeader,
+        retArray);
+  }
+
+  @Test
+  public void testCountIfAlignByDevice() {
+    String[] expectedHeader =
+        new String[] {
+          DEVICE,
+          "Count_if(s1 = 0 & s2 = 0, 3, \"ignoreNull\"=\"true\")",
+          "Count_if(s1 = 1 & s2 = 0, 3)"
+        };
+    String[] retArray = new String[] {"root.db.d1,2,1,"};
+    resultSetEqualTest(
+        "select Count_if(s1=0 & s2=0, 3, 'ignoreNull'='true'), Count_if(s1=1 & s2=0, 3) from root.db.d1 align by device",
+        expectedHeader,
+        retArray);
+  }
+
+  @Test
+  public void testCountIfInHaving() {
+    String[] expectedHeader =
+        new String[] {
+          "Count_if(root.db.d1.s1 = 0 & root.db.d1.s2 = 0, 3, \"ignoreNull\"=\"true\")"
+        };
+    String[] retArray = new String[] {};
+    resultSetEqualTest(
+        "select Count_if(s1=0 & s2=0, 3, 'ignoreNull'='true') from root.db.d1 having Count_if(s1=1 & s2=0, 3) > 1",
```

Comment:
will the having expression share the same column as select if they are same?

Reply:
In having, Count_if(s1=1 & s2=0, 3) > 1 will be transformed to Count_if(s1=1, 3) > 1  & Count_if(s2=0, 3) > 1 now, it does not meet expectations. We need to discuss with @liuminghui233

Reply:
It's a little bit strange, will the `having sum(s1 =1 & s2 >=1) > 100` be translated to `having sum(s1 = 1) > 100 & sum(s2 > 1) > 100`?